cmd/openshift/operator/kodata/tekton-addon/addons/02-clustertasks/source_local/buildah/buildah-task.yaml

```diff
@@ -76,7 +77,25 @@ spec:
         -f $(params.DOCKERFILE) -t $(params.IMAGE) $(params.CONTEXT)
 
       [[ "$(params.SKIP_PUSH)" == "true" ]] && echo "Push skipped" && exit 0
-      [[ "$(workspaces.dockerconfig.bound)" == "true" ]] && export DOCKER_CONFIG="$(workspaces.dockerconfig.path)"
+
+      if [[ "$(workspaces.dockerconfig.bound)" == "true" ]]; then
+
+        # if config.json exists at workspace root, we use that
+        if test -f "$(workspaces.dockerconfig.path)/config.json"; then
+          export DOCKER_CONFIG="$(workspaces.dockerconfig.path)"
+
+        # else we look for .dockerconfigjson at the root
+        elif test -f "$(workspaces.dockerconfig.path)/.dockerconfigjson"; then
+          cp "$(workspaces.dockerconfig.path)/.dockerconfigjson" "$HOME/.docker/config.json"
+          export DOCKER_CONFIG="$HOME/.docker"
+
+        # need to error out if neither files are present
```

Comment:
should we error if nothing is present?

Reply:
I think so. A secret was passed (otherwise we wouldn't go into this condition), but without any valid "data field", so it's probably better to fail "loudly and fast" than silently 🙃

Reply:
well, this is inside
```
if [[ "$(workspaces.dockerconfig.bound)" == "true" ]]; 
```
so the user has supplied a workspace that contains their docker config we couldn't find it, so i think we should be erroring out

Reply:
ah, vincent already replied :P

Reply:
ack